test/traffic_matcher.cc

```diff
@@ -94,12 +94,19 @@ namespace {
       boost::property_tree::read_json(json_ss, answer);
 
       const auto& a_segs = test_case.second;
-      const auto& b_segs = matcher.segments;
+      auto& b_segs = matcher.segments;
+
+      //TODO: for portions of routes with no ots coverage add matching fake segments
+      //for now we just remove the fake ones from the result
+      auto seg_itr = std::remove_if(b_segs.begin(), b_segs.end(),
+        [](const meili::traffic_segment_t& ots){ return !ots.segment_id.Is_Valid(); });
+      b_segs.erase(seg_itr, b_segs.end());
```

Comment:
this is the only substantial change, we delete the segments with invalid ids so they arent part of the test. considering that we are relying on them we should probably add those in at some point